### ISSUE TITLE
Storing access_token/refresh_token relationship.

### ISSUE THREAD

**Issue by red-crown** [bshaffer/oauth2-server-php#520]
It would be nice to able to store the relationship between access_tokens and their refresh_tokens to make revoking tokens and clearing expired tokens cleaner.

However, adding the parameter to RefreshTokenInterface::setRefreshToken would break bwc.

**Comment by pjcdawkins** [comment]
Agreed, it would be useful to implement this, for revocation of refresh tokens and their associated access tokens together.

See RFC 7009 https://tools.ietf.org/html/rfc7009
> If the particular token is a refresh token and the authorization server supports the revocation of access tokens, then the authorization server SHOULD also invalidate all access tokens based on the same authorization grant (see Implementation Note). If the token passed to the request is an access token, the server MAY revoke the respective refresh token as well.

Regarding BC, I think that problem could be overcome. A new interface could be provided, for example (just a thought):

```php
namespace OAuth2\Storage;

interface HierarchicalTokenInterface
{
  public function setParent($token);
  public function getParent($token);
  public function getChildren($token);
}
```
